src/Material/Tooltip.elm

```diff
@@ -0,0 +1,523 @@
+                marginLeft
+          }
+  in
+    out
+
+
+{-| Component update.
+-}
+update : Msg -> Model -> ( Model, Cmd Msg )
+update action model =
+  case action of
+    Enter dom ->
+      ( { model | isActive = True, domState = dom }, none )
+
+    Leave ->
+      ( { model | isActive = False }, none )
+
+
+{-| Tries and get the next sibling that is available and use the given decoder on it
+-}
+sibling : Json.Decoder a -> Json.Decoder a
+sibling d =
+  let
+    createPath depth =
+      let
+        parents =
+          List.repeat depth "parentElement"
+      in
+        ([ "target" ] ++ parents ++ [ "nextSibling" ])
+
+    paths =
+      List.map createPath [0..4]
+
+    -- Tries to check if the element is actually a tooltip
+    valid path =
+      isTooltipClass path
+        `Json.andThen`
+          (\res ->
+            if res then
+              at path d
+            else
+              Json.fail ""
+          )
+  in
+    Json.oneOf (List.map valid paths)
+
+
+{-| Checks if the target at path is an actual tooltip
+-}
+isTooltipClass : List (String) -> Json.Decoder Bool
+isTooltipClass path =
+  (at path DOM.className)
+    `Json.andThen`
+      (\class ->
+        if String.contains "mdl-tooltip" class then
+          Json.succeed True
+        else
+          Json.succeed False
+      )
+
+
+{-| Decodes a DOMState from a DOM event
+-}
+stateDecoder : Json.Decoder DOMState
+stateDecoder =
+  Json.object3 DOMState
+    (DOM.target DOM.boundingClientRect)
+    (sibling DOM.offsetWidth)
+    (sibling DOM.offsetHeight)
+
+
+
+-- PROPERTIES
+
+
+{-| Tooltip size
+-}
+type Size
+  = Default
+  | Large
+
+
+{-| Tooltip position relative to the element
+-}
+type Position
+  = Left
+  | Right
+  | Top
+  | Bottom
+
+
+{-| Tooltip config
+-}
+type alias Config =
+  { size : Size
+  , position : Position
+  }
+
+
+{-| Default configuration for tooltip
+-}
+defaultConfig : Config
+defaultConfig =
+  { size = Default
+  , position = Bottom
+  }
+
+
+{-| Properties for Tooltip options.
+-}
+type alias Property m =
+  Options.Property Config m
+
+
+{-| Position the tooltip on the left of the target element
+-}
+left : Property m
+left =
+  Options.set (\options -> { options | position = Left })
+
+
+{-| Position the tooltip on the right of the target element
+-}
+right : Property m
+right =
+  Options.set (\options -> { options | position = Right })
+
+
+{-| Position the tooltip above the target element
+-}
+top : Property m
+top =
+  Options.set (\options -> { options | position = Top })
+
+
+{-| Position the tooltip below the target element
+-}
+bottom : Property m
+bottom =
+  Options.set (\options -> { options | position = Bottom })
+
+
+{-| Tooltip with the default size
+-}
+default : Property m
+default =
+  Options.set (\options -> { options | size = Default })
+
+
+{-| Large tooltip
+-}
+large : Property m
+large =
+  Options.set (\options -> { options | size = Large })
+
+
+{-| Maps a Html.Attribute to Material Property

```

Comment:
Do we need this? I couldn't find a use of it, and I really don't want to give users the ability to install arbitrary attributes on elm-mdl constructs—I've seen them in the past use it in weird ways that break various things, then file bug reports here.